### PR TITLE
US146673: Pretty-print the JSON in Cloud Foundry app logs

### PR DESCRIPTION
**Pull request by anubhavi25**

### CODE REVIEW

File: utils/pretty-print-cf-app-logs.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+python -c "
+import sys, re, json
+
```
Comment (FrankGasparovic) [inline utils/pretty-print-cf-app-logs.sh]: Maybe add switches for for including router logs in the output.

File: utils/pretty-print-cf-app-logs.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+app_json_regex = r'^(?:-?(?:[1-9][0-9]*)??[0-9]{4})-(?:1[0-2]|0[1-9])-(?:3[01]|0[1-9]|[12][0-9])T(?:2[0-3]|[01][0-9]):(?:[0-5][0-9]):(?:[0-5][0-9])(?:\.[0-9]+)??(?:Z|[+-](?:2[0-3]|[01][0-9])[0-5][0-9])??\s*?\[.*?\](?:\s*?[A-Z]+)??(?:\s+?)??(\{.*\})$'
+bold='\033[1m'
+normal='\033[0m'
+
```
Comment (FrankGasparovic) [inline utils/pretty-print-cf-app-logs.sh]: Consider adding the instance number to the log output.

**Review by FrankGasparovic — DISMISSED** [review]

File: utils/pretty-print-cf-app-logs.sh (hunk 2 above)
Comment (anubhavi25) [reply utils/pretty-print-cf-app-logs.sh]: Done

File: utils/pretty-print-cf-app-logs.sh (hunk 1 above)
Comment (anubhavi25) [reply utils/pretty-print-cf-app-logs.sh]: Done

**Comment by sanjeevchopra** [comment]
👍 @anubhavi25 

We should move this script to a common repo. This will be useful for devs of any service, using the common log format. lets discuss.

Here's some improvements, which will be useful:
* make app/rtr param optional, by default show all logs (apps and rtr)
* change app log format to a single line format, with stack trace following the first line (one line per stack element)
```
2017-07-10T20:59:48.923+0000 ERROR - c089b940-0626-4d0b-5817-3f2d1229d7d2 Failure during POST /v1/client
    "error=\"access_denied\", error_description=\"Error requesting access token.\"",
    "\tat org.springframework.security.oauth2.client.token.OAuth2AccessTokenSupport.retrieveToken(OAuth2AccessTokenSupport.java:145)",
    "\tat org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsAccessTokenProvider.obtainAccessToken(ClientCredentialsAccessTokenProvider.java:44)",
    "\tat org.springframework.security.oauth2.client.token.AccessTokenProviderChain.obtainNewAccessTokenInternal(AccessTokenProviderChain.java:148)",
    "\tat org.springframework.security.oauth2.client.token.AccessTokenProviderChain.obtainAccessToken(AccessTokenProviderChain.java:121)",
    "\tat org.springframework.security.oauth2.client.OAuth2RestTemplate.acquireAccessToken(OAuth2RestTemplate.java:221)",
    "\tat org.springframework.security.oauth2.client.OAuth2RestTemplate.getAccessToken(OAuth2RestTemplate.java:173)",
    "\tat org.springframework.security.oauth2.client.OAuth2RestTemplate.createRequest(OAuth2RestTemplate.java:105)",
    "\tat org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:648)"
     ...
```
* No need to add the 'Router' prefix for RTR logs - it is evident from the log
* if CF header doesnt exist, assume its a app log - this will be useful when running services locally outside cf

(Cc: @Bitty9 @sabertiger)

**Comment by anubhavi25** [comment]
@sanjeevchopra your feedback has been incorporated (script changes are on `develop`). You can still print logs in a multiline format by providing the additional `--multiline` flag.

**Pull request merged**